### ISSUE TITLE
FormatAutodetectionError: "No suitable formats" when passing a json format as a string.

### ISSUE THREAD

**Issue by tigger0jk** [tkarabela/pysubs2#94]
Issue happens for me on 1.7.1 but not 1.6.1 - I downgraded for now but didn't have too much time to look into why.

Here's a small example that fails for me on 1.7.1
```
import pysubs2

# example srt from https://www.speechpad.com/captions/srt
srt_string = """1
00:00:00,498 --> 00:00:02,827
- Here's what I love most
about food and diet.

2
00:00:02,827 --> 00:00:06,383
We all eat several times a day,
and we're totally in charge

3
00:00:06,383 --> 00:00:09,427
of what goes on our plate
and what stays off."""

# this is fine
subs = pysubs2.SSAFile.from_string(srt_string)

json_string = subs.to_string("json")

# this seems fine to me - but I'm not familiar with the format
print(json_string)

# this throws an exception: pysubs2.exceptions.FormatAutodetectionError: No suitable formats
subs = pysubs2.SSAFile.from_string(json_string) # in my real use case even passing format="json" here didn't help - but I didn't have a chance to confirm it doesn't work in this simple example, I think it still fails with the same error.

# we never get here
print(subs.to_string("srt"))
```

```
{"info": {"WrapStyle": "0", "ScaledBorderAndShadow": "yes", "Collisions": "Normal"}, "styles": {"Default": {"fontname": "Arial", "fontsize": 20.0, "primarycolor": {"r": 255, "g": 255, "b": 255, "a": 0}, "secondarycolor": {"r": 255, "g": 0, "b": 0, "a": 0}, "tertiarycolor": {"r": 0, "g": 0, "b": 0, "a": 0}, "outlinecolor": {"r": 0, "g": 0, "b": 0, "a": 0}, "backcolor": {"r": 0, "g": 0, "b": 0, "a": 0}, "bold": false, "italic": false, "underline": false, "strikeout": false, "scalex": 100.0, "scaley": 100.0, "spacing": 0.0, "angle": 0.0, "borderstyle": 1, "outline": 2.0, "shadow": 2.0, "alignment": 2, "marginl": 10, "marginr": 10, "marginv": 10, "alphalevel": 0, "encoding": 1, "drawing": false}}, "events": [{"start": 498, "end": 2827, "text": "- Here's what I love most\\Nabout food and diet.", "marked": false, "layer": 0, "style": "Default", "name": "", "marginl": 0, "marginr": 0, "marginv": 0, "effect": "", "type": "Dialogue"}, {"start": 2827, "end": 6383, "text": "We all eat several times a day,\\Nand we're totally in charge", "marked": false, "layer": 0, "style": "Default", "name": "", "marginl": 0, "marginr": 0, "marginv": 0, "effect": "", "type": "Dialogue"}, {"start": 6383, "end": 9427, "text": "of what goes on our plate\\Nand what stays off.", "marked": false, "layer": 0, "style": "Default", "name": "", "marginl": 0, "marginr": 0, "marginv": 0, "effect": "", "type": "Dialogue"}]}
Traceback (most recent call last):
  File "~/test_pysubs.py", line 28, in <module>
    subs = pysubs2.SSAFile.from_string(json_string)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/venv/lib64/python3.11/site-packages/pysubs2/ssafile.py", line 147, in from_string
    return cls.from_file(fp, format_, fps=fps, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/venv/lib64/python3.11/site-packages/pysubs2/ssafile.py", line 184, in from_file
    format_ = autodetect_format(fragment)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/venv/lib64/python3.11/site-packages/pysubs2/formats/__init__.py", line 78, in autodetect_format
    raise FormatAutodetectionError(content=content, formats=[])
pysubs2.exceptions.FormatAutodetectionError: No suitable formats
```

**Comment by tkarabela** [comment]
@tigger0jk Thanks for the report! I've fixed it in version 1.7.2 :)

**Comment by tigger0jk** [comment]
Thanks! Upgraded to latest and confirmed it works for my real use case. I had looked at the commit that introduced it and didn't see the problem until seeing the commit that fixed it - colon _after_ the quote in `"info":` - appreciate the quick turnaround!
